### PR TITLE
Autogenerate SDK TypeScript definitions on releases

### PR DESCRIPTION
**Pull request by mondeja**
Currently, the type definitions of the SDK function comments are slightly outdated. This PR introduces a script to update the Typescript type definitions generated from the file *sdk.mjs*, executing it on releases.

### CODE REVIEW

File: scripts/release/update-sdk-ts-defs.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  const tempSdkMjsContent = originalSdkMjsContent
+    .split('\n')
+    .filter((line) => {
+      return !line.startsWith(' * @typedef {import("./sdk")');
+    })
+    .join('\n');
+  fs.writeFileSync(sdkMjs, tempSdkMjsContent);
+  try {
+    execSync(
+      'tsc sdk.mjs --declaration --emitDeclarationOnly --allowJs --removeComments',
```
Comment (LitoMore) [inline scripts/release/update-sdk-ts-defs.js:32]:
```suggestion
      'npx tsc sdk.mjs --declaration --emitDeclarationOnly --allowJs --removeComments',
```

The `typescript` didn't install globally. We should use `npx` to run `tsx`.

File: scripts/release/update-sdk-ts-defs.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview
+ * Updates the SDK Typescript definitions located in the file sdk.d.ts
+ * to match the current definitions of functions of sdk.mjs.
+ */
+
+import fs from 'node:fs';
```
Comment (LitoMore) [inline scripts/release/update-sdk-ts-defs.js:7]:
```suggestion
import fs from 'node:fs/promises';
```

We can delightfully use [File System Promises API](https://nodejs.org/api/fs.html#promises-api) and top-level-await.

File: scripts/release/update-sdk-ts-defs.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    console.log(
+      `Error ${error.status} generating Typescript` +
+        ` definitions: '${error.message}'`,
+    );
+    process.exit(1);
+  }
+  await fs.writeFile(sdkMjs, originalSdkMjsContent);
+};
+
+(async () => {
```
Comment (LitoMore) [inline scripts/release/update-sdk-ts-defs.js:46]: This `(async () => { /* code */ })();` wrapper can be removed. [Top-level await](https://www.stefanjudis.com/today-i-learned/top-level-await-is-available-in-node-js-modules/) is available.

File: scripts/release/update-sdk-ts-defs.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  const autogeneratedMsg = '/* The next code is autogenerated from sdk.mjs */';
+  const newSdkTsContent =
+    (await fs.readFile(sdkTs, 'utf-8')).split(autogeneratedMsg)[0] +
+    `${autogeneratedMsg}\n\n${await fs.readFile(sdkMts, 'utf-8')}`;
+
+  await fs.writeFile(sdkTs, newSdkTsContent);
+  await fs.unlink(sdkMts);
+
+  try {
+    execSync('prettier -w sdk.d.ts');
```
Comment (LitoMore) [inline scripts/release/update-sdk-ts-defs.js:59]:
```suggestion
    execSync('npx prettier -w sdk.d.ts');
```

Ditto. We need `npx` to execute `prettier`.

**Review by LitoMore — APPROVED** [review]
We can trigger the `create-release` workflow manually at https://github.com/simple-icons/simple-icons/actions/workflows/create-release.yml.

Simply use the `Run workflow` button.

image

**Review by Gdakgdak22 — APPROVED** [review]
Gracias

**Pull request merged**